pkg/deploy/registry/deployconfig/strategy.go

```diff
@@ -39,14 +41,53 @@ func (strategy) AllowUnconditionalUpdate() bool {
 
 // PrepareForCreate clears fields that are not allowed to be set by end users on creation.
 func (strategy) PrepareForCreate(obj runtime.Object) {
-	_ = obj.(*api.DeploymentConfig)
-	// TODO: need to ensure status.latestVersion is not set out of order
+	dc := obj.(*api.DeploymentConfig)
+	dc.Generation = 1
+	dc.Status = api.DeploymentConfigStatus{}
 }
 
 // PrepareForUpdate clears fields that are not allowed to be set by end users on update.
 func (strategy) PrepareForUpdate(obj, old runtime.Object) {
-	_ = obj.(*api.DeploymentConfig)
-	// TODO: need to ensure status.latestVersion is not set out of order
+	newDc := obj.(*api.DeploymentConfig)
+	oldDc := old.(*api.DeploymentConfig)
+
+	newVersion := newDc.Status.LatestVersion
+	oldVersion := oldDc.Status.LatestVersion
+
+	// Check for imagechange controller change
+	isImageControllerChange := deployutil.IsImageChangeControllerChange(*newDc, *oldDc)
+
+	// Persist status
+	newStatus := newDc.Status
+	newDc.Status = oldDc.Status
+
+	// If this update comes from the imagechange controller, tolerate status detail updates.
+	// TODO: Make the config change controller detect this change and update latestVersion.
+	// Then the main controller should detect there is a change coming from the image change
+	// controller and update status details accordingly.
+	if isImageControllerChange {
+		newDc.Status.Details = newStatus.Details
+	}
+
+	// oc deploy --latest from new clients
+	if newVersion == oldVersion && deployutil.IsInstantiated(newDc) {
+		// TODO: Have an endpoint for this and avoid hacking it here.
```

Comment:
I had been running without a problem on it when I was bumping generation. Also tests have never failed me. I am going to spin up a cluster and do some stress-testing.